### ISSUE TITLE
crash in CacheFile::Copy

### ISSUE THREAD

**Issue by jaharkes** [cmusatyalab/coda#103]
Seems VASTRO related. We crashed  when trying to copy a bitmap while copying a cachefile during repair. Why we have a bitmap to copy, I'm not sure because this was a CML_STORE operation so it couldn't have been a partially cache file. Either way we crashed because the recoverable bitmaps require a transaction.

CacheFile::Copy is called from only a few places.
 - local_cml.cc during preservelocal repair when we're copying from the locally cached copy to the 'global copy' of a file. (currently called outside of a transaction)
- fso1.cc during cache recovery when an fsobj fails to recover, we back up the content to a spool file (no transaction)
- fso1.cc when creating a shadow copy (fsobj::MakeShadow)

fsobj::MakeShadow is called from
- venuscb.cc CallBackFetch making sure the object we are reintegration is not modified during reintegration. This isn't actively used because we are only using partial reintegration for stores. (no transaction)
- vol_cml.cc cmlent:Freeze and cmlent::AttachFidBindings when creating a shadow copy when starting reintegration (called with active transaction)

So basically we are sometimes called without a transaction, and sometimes with, but most cases we probably don't care, or even shouldn't have to care about partially cached files because except for cache recovery during startup all are in the file write paths or an extension thereof.
